lib/marked.js

```diff
@@ -730,6 +730,14 @@ InlineLexer.prototype.output = function(src) {
       } else if (this.inLink && /^<\/a>/i.test(cap[0])) {
         this.inLink = false;
       }
+      if (!this.inRawBlock
+          && /^<pre|^<code|^<kbd|^<script/i.test(cap[0])) {
+        this.inRawBlock = true;
+      } else if (this.inRawBlock
+          && /^<\/pre>|^<\/code>|^<\/kbd>|^<\/script>/i.test(cap[0])) {
+        this.inRawBlock = false;
+      }
```

Comment:
There's some duplication here: it'd be nice to just have a simple list of tags in which no substitution is to be performed -- `['pre', 'code', 'kbd', 'script']` -- 'cos that'd make it easy to add/remove tags; I did try that route but the regexes looked less "on the nose" than this version.

I also experimented with alternate names for the `inRawBlock` field: they were all too darn long and made the code less readable: forex, `isBlockInWhichNoSubstitutionIsToBePerformed`. Happy to take a steer here 😄

Reply:
Could you change them to `/^<(pre|code|kbd|script)/i` and `/^<\/(pre|code|kbd|script)>/i`

I think that makes it a little easier to see what the regex is looking for